go/pkg/pwcompose/compose.go

```diff
@@ -138,12 +138,60 @@ func Up(preparedCompose string, instanceKey string, logger *zap.Logger) error {
 		zap.String("compose", preparedCompose),
 		zap.String("instance-key", instanceKey),
 	)
-	// parse compose labels to get flavor info
-	// generate instanceID with flavor+instanceKey
-	// exec docker-compose up -d with params
-	// print instanceID
-	// later: print instance passphrases
-	return fmt.Errorf("not implemented")
+
+	// parse prepared compose yaml
+	preparedComposeStruct := config{}
+	err := yaml.Unmarshal([]byte(preparedCompose), &preparedComposeStruct)
+	if err != nil {
+		return fmt.Errorf("parse prepared compose: %w", err)
+	}
+
+	// generate instanceIDs and set them as container_name
+	for name, service := range preparedComposeStruct.Services {
+		service.ContainerName = service.Image + "." + instanceKey
+		preparedComposeStruct.Services[name] = service
+	}
+
+	// create tmp docker-compose file
+	tmpData, err := yaml.Marshal(&preparedComposeStruct)
+	if err != nil {
+		return fmt.Errorf("marshal config: %w", err)
+	}
+	tmpFile, err := os.Create("docker-compose.tmp.yml")
```

Comment:
can you create a temporary file or temporary directory to avoid issues when running multiple `pathwar compose up` in parallel?

Reply:
done